implementations/rust/ockam/ockam_command/src/enroll/oidc_service.rs

```diff
@@ -520,4 +580,66 @@ mod tests {
         assert!(code.is_ok());
         assert_eq!(code.unwrap(), "12345".to_string())
     }
+
+    enum ClientState {
+        FirstCall,
+        SecondCall,
+        Finished
+    }
+
+    struct Client {
+        state: Arc<Mutex<ClientState>>
+    }
+
+    #[async_trait]
+    impl GetUserInfo for Client{
+        async fn get_user_info(&self, _token: &OidcToken) -> Result<UserInfo> {
+
+                let mut guard = self.state.lock().unwrap();
+
+            match *guard {
+                ClientState::FirstCall => {
+                     *guard = ClientState::SecondCall;
+                    return Ok(UserInfo{
+                        sub: "".to_string(),
+                        nickname: "bad_nickname".to_string(),
+                        name: "".to_string(),
+                        picture: "".to_string(),
+                        updated_at: "".to_string(),
+                        email: "".to_string(),
+                        email_verified: false,
+                    })
+                }
+
+                ClientState::SecondCall => {
+                    *guard = ClientState::Finished;
+                    return Ok(UserInfo{
+                        sub: "".to_string(),
+                        nickname: "my_cool_nickname".to_string(),
+                        name: "".to_string(),
+                        picture: "".to_string(),
+                        updated_at: "".to_string(),
+                        email: "".to_string(),
+                        email_verified: true,
+                    })
+                }
+
+                ClientState::Finished => panic!("an extra call!")
+            }
+        }
+    }
+
+    #[test] fn test_wait_for_email_verification() -> Result<()>{
```

Comment:
that's basically happens because when we create the ops `let opts = CommandGlobalOpts::new(GlobalArgs::default()).set_quiet();` it calls a [executor](https://github.com/build-trust/ockam/blob/772eb8cb6aafbb46e3be0403a3cf22e2ff015993/implementations/rust/ockam/ockam_api/src/cli_state.rs#L109). So if we make the function async, it throws a error saying we cannot create a runtime within another runtime.

Reply:
Right, I've had this kind of issue in the past :-). One solution is to just pass a `Terminal` to `wait_for_email_verification` since that function does not need a full set of global options.